universal-application-tool-0.0.1/app/models/Program.java

```diff
@@ -0,0 +1,28 @@
+package models;
+
+import io.ebean.annotation.DbJsonB;
+import java.util.Map;
+import javax.persistence.Entity;
+import play.data.validation.Constraints;
+
+@Entity
+/** The ebeans mapped class for the program object. */
+public class Program {
+  private static final long serialVersionUID = 1;
+
+  @Constraints.Required @DbJsonB
+  // When we build an object that Jackson can deserialize, we replace Map<String, Object> with that
+  // type.
+  // For now, this will be automatically deserialized - with subobjects being "Map<String, Object>"
+  // and lists
+  // being List<Object>.
+  public Map<String, Object> object;
```

Comment:
Suggestion: make these variables package-private and access them with getters/setters

Reply:
Thoughts: EBean automatically generates getters and setters for us. However, those cannot be used by our code since the generation happens at a later stage after our code is compiled (I might be wrong about this though.)

Not against it, just feel there will be a lot of boilerplate code for these if we choose to write our own.

Reply:
The downside of accessing instance variables directly from outside the class is that we have to refactor all call sites if we need to add behavior to getting/setting the variable.

Reply:
More on this: go/java-practices/visibility#public_fields

Reply:
Eh, it's easy to do - I can go press the button to generate getters and setters in IntelliJ - I'll do that now.